### PR TITLE
Improve sequence values in style

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17530)

### CODE REVIEW

**Comment by nox** [comment]
r? @emilio 
@bors-servo try
https://treeherder.mozilla.org/#/jobs?repo=try&revision=da8a3e76ad15461db6aee4a544b6fc23dfa07362

**Comment by nox** [comment]
Had not made the generated `ToCss` impl use `Separator::separator()`. 

https://treeherder.mozilla.org/#/jobs?repo=try&revision=ecb9d5e1dacfcdac521d53842b7a50226df96449

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -3474,12 +3478,13 @@ fn static_assert() {
             gecko_filter.mFilterParameter.set_value(value);
         }
 
+        let v = v.into_iter();
         unsafe {
-            Gecko_ResetFilters(&mut self.gecko, v.0.len());
+            Gecko_ResetFilters(&mut self.gecko, v.len());
         }
-        debug_assert!(v.0.len() == self.gecko.mFilters.len());
+        debug_assert!(v.len() == self.gecko.mFilters.len());
```
Comment (emilio) [inline components/style/properties/gecko.mako.rs]: while you're here, you can change this to `debug_assert_eq!` if you want.

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 2]
```diff
@@ -3340,11 +3340,13 @@ impl Animatable for AnimatedFilterList {
...
         }
 
-        Ok(filters.into())
+        Ok(AnimatedFilterList(filters))
     }
 
     fn add(&self, other: &Self) -> Result<Self, ()> {
-        Ok(self.0.iter().chain(other.0.iter()).cloned().collect::<Vec<_>>().into())
+        Ok(AnimatedFilterList(
+            self.0.iter().chain(other.0.iter()).cloned().collect::<Vec<_>>(),
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: Do you really need the type annotation?

File: components/style_traits/values.rs [hunk 3]
```diff
@@ -191,6 +191,12 @@ pub struct Comma;
 /// separated by spaces.
 pub struct Space;
 
+/// Type used as the associated type in the `OneOrMoreSeparated` trait on a
+/// type to indicate that a serialized list of elements of this type is
+/// separated by commas, but spaces without commas are also allowed when
+/// parsing.
+pub struct CommaWithSpace;
```
Comment (emilio) [inline components/style_traits/values.rs]: `CommaOrWhitespace`, maybe? (no big deal anyway)

File: components/style_traits/values.rs [hunk 4]
```diff
@@ -240,14 +246,41 @@ impl Separator for Space {
     where
         F: for<'tt> FnMut(&mut Parser<'i, 'tt>) -> Result<T, ParseError<'i, E>>
     {
-        let mut results = vec![];
+        let mut results = vec![parse_one(input)?];
```
Comment (emilio) [inline components/style_traits/values.rs]: This should be in the previous commit, right?

**Review by emilio — APPROVED** [review]

**Pull request merged**